--- /root/.pyenv/versions/3.11.11/lib/python3.11/site-packages/transformers/models/qwen2/modeling_qwen2.py	2025-05-02 15:36:13.533609384 +0000
+++ /develop/TensorRT/patches/eagle2/modeling_qwen2.py	2025-05-02 15:25:35.584592474 +0000
@@ -593,7 +593,7 @@
                         " this may lead to unexpected behaviour for Flash Attention version of Qwen2. Make sure to "
                         " call `tokenizer.padding_side  = 'left'` before tokenizing the input. "
                     )
-            if attention_mask is not None and 0.0 in attention_mask:
+            if attention_mask is not None: # and 0.0 in attention_mask:
                 return attention_mask
             return None
 
