--- /opt/conda/lib/python3.8/site-packages/pytorch_quantization/nn/modules/tensor_quantizer.py	2021-08-16 22:50:37.000000000 +0000
+++ tensor_quantizer.py	2021-10-19 20:41:54.288077426 +0000
@@ -291,7 +291,7 @@
             quant_dim = list(amax.shape).index(list(amax_sequeeze.shape)[0])
             scale = amax_sequeeze / bound
             outputs = torch.fake_quantize_per_channel_affine(
-                inputs, scale.data, torch.zeros_like(scale, dtype=torch.long).data, quant_dim,
+                inputs, scale.data, torch.zeros_like(scale, dtype=torch.int32).data, quant_dim,
                 -bound - 1 if not self._unsigned else 0, bound)

         return outputs
