diff --git a/core/conversion/converters/impl/activation.cpp b/core/conversion/converters/impl/activation.cpp
index 64edeed..5279413 100644
--- a/core/conversion/converters/impl/activation.cpp
+++ b/core/conversion/converters/impl/activation.cpp
@@ -152,21 +152,6 @@ auto acthardtanh TRTORCH_UNUSED =
                     out_tensor = ctx->AssociateValueAndTensor(n->outputs()[0], out_tensor);
                     LOG_DEBUG("Output shape: " << out_tensor->getDimensions());
                     return true;
-                  }})
-        .pattern({"aten::elu(Tensor self, Scalar alpha=1, Scalar scale=1, Scalar input_scale=1) -> (Tensor)",
-                  [](ConversionCtx* ctx, const torch::jit::Node* n, args& args) -> bool {
-                    auto in = args[0].ITensorOrFreeze(ctx);
-                    auto alpha = args[1].unwrapToDouble();
-
-                    auto new_layer = ctx->net->addActivation(*in, nvinfer1::ActivationType::kELU);
-                    TRTORCH_CHECK(new_layer, "Unable to create layer for aten::elu");
-                    new_layer->setAlpha(alpha);
-
-                    new_layer->setName(trtorch::core::util::node_info(n).c_str());
-
-                    auto out_tensor = ctx->AssociateValueAndTensor(n->outputs()[0], new_layer->getOutput(0));
-                    LOG_DEBUG("Output shape: " << out_tensor->getDimensions());
-                    return true;
                   }});

 } // namespace
